### ISSUE TITLE
Change the name field of bigip_ltm_profile_client_ssl causes provider fatal error

### ISSUE THREAD

**Issue by AliAllomani** [F5Networks/terraform-provider-bigip#505]
### Environment

 * TMOS/Bigip Version:
 * Terraform Version: 0.12.6
 * Terraform bigip provider Version: v1.1.1

### Summary

Change the name field of bigip_ltm_profile_client_ssl causes provider fatal error

This is mostly because the name filed is used as ID of the resource and there is no `ForceNew` defined for the name field

https://github.com/F5Networks/terraform-provider-bigip/blob/98fbadf10d89f9908b7bd4b4bcad6e4c658fd48a/bigip/resource_bigip_ltm_profile_ssl_client.go#L27-L33

https://github.com/F5Networks/terraform-provider-bigip/blob/98fbadf10d89f9908b7bd4b4bcad6e4c658fd48a/bigip/resource_bigip_ltm_profile_ssl_client.go#L501-L505

https://github.com/F5Networks/terraform-provider-bigip/blob/98fbadf10d89f9908b7bd4b4bcad6e4c658fd48a/bigip/resource_bigip_ltm_profile_ssl_client.go#L510

Plan

```hcl
    ~ resource "bigip_ltm_profile_client_ssl" "client_ssl" {
...
        ~ name                                = "/Common/c_ssl_tftesting_9000" -> "/Common/c_ssl_tftesting_9000_428938605"
...
      }
      
```

Apply
```hcl

Error: Provider produced inconsistent final plan

When expanding the plan for module.lb.bigip_ltm_virtual_server.vs["9000"] to
include new values learned so far during apply, provider "bigip" produced an
invalid new value for .client_profiles: planned set element
cty.StringVal("/Common/c_ssl_tftesting_9000_428938605") does not correlate
with any element in actual.

This is a bug in the provider, which should be reported in the provider's own
issue tracker.

```

Seems the same also applies to `bigip_ltm_profile_server_ssl`, but i did not test it

### Steps To Reproduce

1. create bigip_ltm_profile_client_ssl resource
2. update bigip_ltm_profile_client_ssl resource with different `name` field

### Expected Behavior

Updating the name field should not cause terraform error

### Actual Behavior
Updating the name field causes 

```
provider "bigip" produced an invalid new value for .client_profiles: planned set element
```

**Comment by RavinderReddyF5** [comment]
@AliAllomani can u please try with latest terraform provider v1.9.0.?

**Comment by AliAllomani** [comment]
Sorry @RavinderReddyF5  i cannot test with the newer version at the moment, but from what i see when comparing the files of v1.1.1 with v1.9.0, there is not change regarding this point

hopefully you can re-produce it on your side.

I think would be helpful to add updating the name to the acceptance test (if it can be updated without ForceNew) as i don't see name update in any of the test cases.


```diff
diff --git a/bigip/resource_bigip_ltm_profile_ssl_client.go b/bigip/resource_bigip_ltm_profile_ssl_client.go
index 818c6537..6f6d3ad9 100644
--- a/bigip/resource_bigip_ltm_profile_ssl_client.go
+++ b/bigip/resource_bigip_ltm_profile_ssl_client.go
@@ -7,10 +7,11 @@ package bigip
 
 import (
 	"fmt"
-	"log"
-
 	"github.com/f5devcentral/go-bigip"
-	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"log"
+	"reflect"
+	"strings"
 )
 
 func resourceBigipLtmProfileClientSsl() *schema.Resource {
@@ -25,26 +26,30 @@ func resourceBigipLtmProfileClientSsl() *schema.Resource {
 
 		Schema: map[string]*schema.Schema{
 			"name": {
-				Type:        schema.TypeString,
-				Required:    true,
-				Description: "Name of the Ssl Profile",
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "Name of the Ssl Profile",
+				ValidateFunc: validateF5NameWithDirectory,
 			},
 
 			"partition": {
 				Type:        schema.TypeString,
 				Optional:    true,
+				Computed:    true,
 				Description: "name of partition",
 			},
 
 			"full_path": {
 				Type:        schema.TypeString,
 				Optional:    true,
+				Computed:    true,
 				Description: "full path of the profile",
 			},
 
 			"generation": {
 				Type:        schema.TypeInt,
 				Optional:    true,
+				Computed:    true,
 				Description: "generation",
 			},
 
@@ -76,6 +81,27 @@ func resourceBigipLtmProfileClientSsl() *schema.Resource {
 				Description: "Client certificate chain traversal depth.  Default 9.",
 			},
 
+			"c3d_client_fallback_cert": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "Client Fallback Certificate. Default None.",
+			},
+
+			"c3d_drop_unknown_ocsp_status": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "Unknown OCSP Response Control. Default Drop.",
+			},
+
+			"c3d_ocsp": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "OCSP. Default None.",
+			},
+
 			"ca_file": {
 				Type:        schema.TypeString,
 				Optional:    true,
@@ -98,10 +124,11 @@ func resourceBigipLtmProfileClientSsl() *schema.Resource {
 			},
 
 			"cert": {
-				Type:        schema.TypeString,
-				Optional:    true,
-				Computed:    true,
-				Description: "Name of the server certificate.",
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validateF5Name,
+				Description:  "Name of the server certificate.",
 			},
 
 			"cert_key_chain": {
@@ -198,7 +225,7 @@ func resourceBigipLtmProfileClientSsl() *schema.Resource {
 			"defaults_from": {
 				Type:        schema.TypeString,
 				Optional:    true,
-				Computed:    true,
+				Default:     "/Common/clientssl",
 				Description: "Profile name that this profile defaults from.",
 			},
 
@@ -231,10 +258,11 @@ func resourceBigipLtmProfileClientSsl() *schema.Resource {
 			},
 
 			"key": {
-				Type:        schema.TypeString,
-				Optional:    true,
-				Computed:    true,
-				Description: "Name of the Server SSL profile key",
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validateF5Name,
+				Description:  "Name of the Server SSL profile key",
 			},
 
 			"mod_ssl_methods": {
@@ -378,6 +406,13 @@ func resourceBigipLtmProfileClientSsl() *schema.Resource {
 				Description: "SNI Require (true / false)",
 			},
 
+			"ssl_c3d": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "Client Certificate Constrained Delegation enabled / disabled.  Default is disabled.",
+			},
+
 			"ssl_forward_proxy": {
 				Type:        schema.TypeString,
 				Optional:    true,
@@ -418,15 +453,34 @@ func resourceBigipLtmProfileClientSsl() *schema.Resource {
 
 func resourceBigipLtmProfileClientSSLCreate(d *schema.ResourceData, meta interface{}) error {
 	client := meta.(*bigip.BigIP)
-
 	name := d.Get("name").(string)
 	parent := d.Get("defaults_from").(string)
-	log.Println("[INFO] Creating Client Ssl Profile " + name)
 
-	err := client.CreateClientSSLProfile(
-		name,
-		parent,
-	)
+	log.Printf("[INFO] Creating Client Ssl Profile:%+v ", name)
+
+	sslForwardProxyEnabled := d.Get("ssl_forward_proxy").(string)
+	inheritCertkeychain := d.Get("inherit_cert_keychain").(string)
+	proxyCaCert := d.Get("proxy_ca_cert").(string)
+	proxyCaKey := d.Get("proxy_ca_key").(string)
+	sslForwardProxyBypass := d.Get("ssl_forward_proxy_bypass").(string)
+	if sslForwardProxyEnabled == "enabled" {
+		proxyCaCert = "/Common/default.crt"
+		proxyCaKey = "/Common/default.key"
+		inheritCertkeychain = "true"
+		if sslForwardProxyBypass == "" {
+			sslForwardProxyBypass = "disabled"
+		}
+	}
+	pss := &bigip.ClientSSLProfile{
+		Name:                  name,
+		DefaultsFrom:          parent,
+		InheritCertkeychain:   inheritCertkeychain,
+		ProxyCaCert:           proxyCaCert,
+		ProxyCaKey:            proxyCaKey,
+		SslForwardProxy:       sslForwardProxyEnabled,
+		SslForwardProxyBypass: sslForwardProxyBypass,
+	}
+	err := client.CreateClientSSLProfile(pss)
 
 	if err != nil {
 		log.Printf("[ERROR] Unable to Create Client Ssl Profile (%s) (%v)", name, err)
@@ -437,7 +491,7 @@ func resourceBigipLtmProfileClientSSLCreate(d *schema.ResourceData, meta interfa
 
 	err = resourceBigipLtmProfileClientSSLUpdate(d, meta)
 	if err != nil {
-		client.DeleteClientSSLProfile(name)
+		_ = client.DeleteClientSSLProfile(name)
 		return err
 	}
 
@@ -446,15 +500,14 @@ func resourceBigipLtmProfileClientSSLCreate(d *schema.ResourceData, meta interfa
 
 func resourceBigipLtmProfileClientSSLUpdate(d *schema.ResourceData, meta interface{}) error {
 	client := meta.(*bigip.BigIP)
-
 	name := d.Id()
 
-	//log.Println("[INFO] Updating Route " + description)
+	log.Printf("[INFO] Updating Clientssl Profile : %v", name)
 
-	//var tmOptions []string
-	//if t, ok := d.GetOk("tm_options"); ok {
-	//	tmOptions = setToStringSlice(t.(*schema.Set))
-	//}
+	var tmOptions []string
+	if t, ok := d.GetOk("tm_options"); ok {
+		tmOptions = setToStringSlice(t.(*schema.Set))
+	}
 
 	var CertExtensionIncludes []string
 	if cei, ok := d.GetOk("cert_extension_includes"); ok {
@@ -488,6 +541,15 @@ func resourceBigipLtmProfileClientSSLUpdate(d *schema.ResourceData, meta interfa
 			Passphrase: d.Get(prefix + ".passphrase").(string),
 		})
 	}
+	sslForwardProxyEnabled := d.Get("ssl_forward_proxy").(string)
+	inheritCertkeychain := d.Get("inherit_cert_keychain").(string)
+	proxyCaCert := d.Get("proxy_ca_cert").(string)
+	proxyCaKey := d.Get("proxy_ca_key").(string)
+	if sslForwardProxyEnabled == "enabled" {
+		proxyCaCert = "/Common/default.crt"
+		proxyCaKey = "/Common/default.key"
+		inheritCertkeychain = "true"
+	}
 
 	pss := &bigip.ClientSSLProfile{
 		Name:                            d.Get("name").(string),
@@ -498,6 +560,9 @@ func resourceBigipLtmProfileClientSSLUpdate(d *schema.ResourceData, meta interfa
 		AllowNonSsl:                     d.Get("allow_non_ssl").(string),
 		Authenticate:                    d.Get("authenticate").(string),
 		AuthenticateDepth:               d.Get("authenticate_depth").(int),
+		C3dClientFallbackCert:           d.Get("c3d_client_fallback_cert").(string),
+		C3dDropUnknownOcspStatus:        d.Get("c3d_drop_unknown_ocsp_status").(string),
+		C3dOcsp:                         d.Get("c3d_ocsp").(string),
 		CaFile:                          d.Get("ca_file").(string),
 		CacheSize:                       d.Get("cache_size").(int),
 		CacheTimeout:                    d.Get("cache_timeout").(int),
@@ -514,38 +579,44 @@ func resourceBigipLtmProfileClientSSLUpdate(d *schema.ResourceData, meta interfa
 		ForwardProxyBypassDefaultAction: d.Get("forward_proxy_bypass_default_action").(string),
 		GenericAlert:                    d.Get("generic_alert").(string),
 		HandshakeTimeout:                d.Get("handshake_timeout").(string),
-		InheritCertkeychain:             d.Get("inherit_cert_keychain").(string),
+		InheritCertkeychain:             inheritCertkeychain,
 		Key:                             d.Get("key").(string),
 		ModSslMethods:                   d.Get("mod_ssl_methods").(string),
 		Mode:                            d.Get("mode").(string),
-		//TmOptions:                       tmOptions,
-		Passphrase:            d.Get("passphrase").(string),
-		PeerCertMode:          d.Get("peer_cert_mode").(string),
-		ProxyCaCert:           d.Get("proxy_ca_cert").(string),
-		ProxyCaKey:            d.Get("proxy_ca_key").(string),
-		ProxyCaPassphrase:     d.Get("proxy_ca_passphrase").(string),
-		ProxySsl:              d.Get("proxy_ssl").(string),
-		ProxySslPassthrough:   d.Get("proxy_ssl_passthrough").(string),
-		RenegotiatePeriod:     d.Get("renegotiate_period").(string),
-		RenegotiateSize:       d.Get("renegotiate_size").(string),
-		Renegotiation:         d.Get("renegotiation").(string),
-		RetainCertificate:     d.Get("retain_certificate").(string),
-		SecureRenegotiation:   d.Get("secure_renegotiation").(string),
-		ServerName:            d.Get("server_name").(string),
-		SessionMirroring:      d.Get("session_mirroring").(string),
-		SessionTicket:         d.Get("session_ticket").(string),
-		SniDefault:            d.Get("sni_default").(string),
-		SniRequire:            d.Get("sni_require").(string),
-		SslForwardProxy:       d.Get("ssl_forward_proxy").(string),
-		SslForwardProxyBypass: d.Get("ssl_forward_proxy_bypass").(string),
-		SslSignHash:           d.Get("ssl_sign_hash").(string),
-		StrictResume:          d.Get("strict_resume").(string),
-		UncleanShutdown:       d.Get("unclean_shutdown").(string),
+		Passphrase:                      d.Get("passphrase").(string),
+		PeerCertMode:                    d.Get("peer_cert_mode").(string),
+		ProxyCaPassphrase:               d.Get("proxy_ca_passphrase").(string),
+		ProxySsl:                        d.Get("proxy_ssl").(string),
+		ProxySslPassthrough:             d.Get("proxy_ssl_passthrough").(string),
+		RenegotiatePeriod:               d.Get("renegotiate_period").(string),
+		RenegotiateSize:                 d.Get("renegotiate_size").(string),
+		Renegotiation:                   d.Get("renegotiation").(string),
+		RetainCertificate:               d.Get("retain_certificate").(string),
+		SecureRenegotiation:             d.Get("secure_renegotiation").(string),
+		ServerName:                      d.Get("server_name").(string),
+		SessionMirroring:                d.Get("session_mirroring").(string),
+		SessionTicket:                   d.Get("session_ticket").(string),
+		SniDefault:                      d.Get("sni_default").(string),
+		SniRequire:                      d.Get("sni_require").(string),
+		SslC3d:                          d.Get("ssl_c3d").(string),
+		SslForwardProxy:                 sslForwardProxyEnabled,
+		SslForwardProxyBypass:           d.Get("ssl_forward_proxy_bypass").(string),
+		SslSignHash:                     d.Get("ssl_sign_hash").(string),
+		StrictResume:                    d.Get("strict_resume").(string),
+		UncleanShutdown:                 d.Get("unclean_shutdown").(string),
+	}
+	if len(tmOptions) > 0 {
+		pss.TmOptions = tmOptions
+	}
+	if proxyCaCert != "none" {
+		pss.ProxyCaCert = proxyCaCert
+	}
+	if proxyCaKey != "none" {
+		pss.ProxyCaKey = proxyCaKey
 	}
-
 	err := client.ModifyClientSSLProfile(name, pss)
 	if err != nil {
-		return fmt.Errorf("Error create profile Ssl (%s): %s", name, err)
+		return fmt.Errorf(" Error create profile Ssl (%s): %s", name, err)
 	}
 	return resourceBigipLtmProfileClientSSLRead(d, meta)
 }
@@ -568,8 +639,8 @@ func resourceBigipLtmProfileClientSSLRead(d *schema.ResourceData, meta interface
 		return nil
 	}
 
-	d.Set("name", name)
-	d.Set("partition", obj.Partition)
+	_ = d.Set("name", name)
+	_ = d.Set("partition", obj.Partition)
 
 	if err := d.Set("defaults_from", obj.DefaultsFrom); err != nil {
 		return fmt.Errorf("[DEBUG] Error saving DefaultsFrom to state for Ssl profile  (%s): %s", d.Id(), err)
@@ -591,6 +662,18 @@ func resourceBigipLtmProfileClientSSLRead(d *schema.ResourceData, meta interface
 		return fmt.Errorf("[DEBUG] Error saving AuthenticateDepth to state for Ssl profile  (%s): %s", d.Id(), err)
 	}
 
+	if err := d.Set("c3d_client_fallback_cert", obj.C3dClientFallbackCert); err != nil {
+		return fmt.Errorf("[DEBUG] Error saving C3dClientFallbackCert to state for Ssl profile  (%s): %s", d.Id(), err)
+	}
+
+	if err := d.Set("c3d_drop_unknown_ocsp_status", obj.C3dDropUnknownOcspStatus); err != nil {
+		return fmt.Errorf("[DEBUG] Error saving C3dDropUnknownOcspStatus to state for Ssl profile  (%s): %s", d.Id(), err)
+	}
+
+	if err := d.Set("c3d_ocsp", obj.C3dOcsp); err != nil {
+		return fmt.Errorf("[DEBUG] Error saving C3dOcsp to state for Ssl profile  (%s): %s", d.Id(), err)
+	}
+
 	if err := d.Set("ca_file", obj.CaFile); err != nil {
 		return fmt.Errorf("[DEBUG] Error saving CaFile to state for Ssl profile  (%s): %s", d.Id(), err)
 	}
@@ -609,12 +692,11 @@ func resourceBigipLtmProfileClientSSLRead(d *schema.ResourceData, meta interface
 
 	for i, c := range obj.CertKeyChain {
 		ckc := fmt.Sprintf("cert_key_chain.%d", i)
-		d.Set(fmt.Sprintf("%s.name", ckc), c.Name)
-		d.Set(fmt.Sprintf("%s.cert", ckc), c.Cert)
-		d.Set(fmt.Sprintf("%s.chain", ckc), c.Chain)
-		d.Set(fmt.Sprintf("%s.key", ckc), c.Key)
-		d.Set(fmt.Sprintf("%s.passphrase", ckc), c.Passphrase)
-
+		_ = d.Set(fmt.Sprintf("%s.name", ckc), c.Name)
+		_ = d.Set(fmt.Sprintf("%s.cert", ckc), c.Cert)
+		_ = d.Set(fmt.Sprintf("%s.chain", ckc), c.Chain)
+		_ = d.Set(fmt.Sprintf("%s.key", ckc), c.Key)
+		_ = d.Set(fmt.Sprintf("%s.passphrase", ckc), c.Passphrase)
 	}
 
 	if err := d.Set("cert_extension_includes", obj.CertExtensionIncludes); err != nil {
@@ -672,9 +754,36 @@ func resourceBigipLtmProfileClientSSLRead(d *schema.ResourceData, meta interface
 		return fmt.Errorf("[DEBUG] Error saving Mode to state for Ssl profile  (%s): %s", d.Id(), err)
 	}
 
-	//if err := d.Set("tm_options", obj.TmOptions); err != nil {
-	//	return fmt.Errorf("[DEBUG] Error saving TmOptions to state for Ssl profile  (%s): %s", d.Id(), err)
-	//}
+	xt := reflect.TypeOf(obj.TmOptions).Kind()
+	if obj.TmOptions != "none" && xt == reflect.String {
+		tmOptions := strings.Split(obj.TmOptions.(string), " ")
+		if len(tmOptions) > 0 {
+			tmOptions = tmOptions[1:]
+			tmOptions = tmOptions[:len(tmOptions)-1]
+		}
+		if err := d.Set("tm_options", tmOptions); err != nil {
+			return fmt.Errorf("[DEBUG] Error saving TmOptions to state for Ssl profile  (%s): %s", d.Id(), err)
+		}
+	} else if obj.TmOptions != "none" && xt != reflect.String {
+		var newObj []string
+		for _, v := range obj.TmOptions.([]interface{}) {
+			newObj = append(newObj, v.(string))
+		}
+		if err := d.Set("tm_options", newObj); err != nil {
+			return fmt.Errorf("[DEBUG] Error saving TmOptions to state for Ssl profile  (%s): %s", d.Id(), err)
+		}
+	} else {
+		tmOptions := []string{}
+		if err := d.Set("tm_options", tmOptions); err != nil {
+			return fmt.Errorf("[DEBUG] Error saving TmOptions to state for Ssl profile  (%s): %s", d.Id(), err)
+		}
+	}
+	if err := d.Set("proxy_ca_cert", obj.ProxyCaCert); err != nil {
+		return fmt.Errorf("[DEBUG] Error saving Mode to state for Ssl profile  (%s): %s", d.Id(), err)
+	}
+	if err := d.Set("proxy_ca_key", obj.ProxyCaKey); err != nil {
+		return fmt.Errorf("[DEBUG] Error saving Mode to state for Ssl profile  (%s): %s", d.Id(), err)
+	}
 
 	if err := d.Set("passphrase", obj.Passphrase); err != nil {
 		return fmt.Errorf("[DEBUG] Error saving Passphrase to state for Ssl profile  (%s): %s", d.Id(), err)
@@ -684,14 +793,6 @@ func resourceBigipLtmProfileClientSSLRead(d *schema.ResourceData, meta interface
 		return fmt.Errorf("[DEBUG] Error saving PeerCertMode to state for Ssl profile  (%s): %s", d.Id(), err)
 	}
 
-	if err := d.Set("proxy_ca_cert", obj.ProxyCaCert); err != nil {
-		return fmt.Errorf("[DEBUG] Error saving ProxySsl to state for Ssl profile  (%s): %s", d.Id(), err)
-	}
-
-	if err := d.Set("proxy_ca_key", obj.ProxyCaKey); err != nil {
-		return fmt.Errorf("[DEBUG] Error saving ProxyCaKey to state for Ssl profile  (%s): %s", d.Id(), err)
-	}
-
 	if err := d.Set("proxy_ca_passphrase", obj.ProxyCaPassphrase); err != nil {
 		return fmt.Errorf("[DEBUG] Error saving ProxyCaPassphrase to state for Ssl profile  (%s): %s", d.Id(), err)
 	}
@@ -744,6 +845,10 @@ func resourceBigipLtmProfileClientSSLRead(d *schema.ResourceData, meta interface
 		return fmt.Errorf("[DEBUG] Error saving SniRequire to state for Ssl profile  (%s): %s", d.Id(), err)
 	}
 
+	if err := d.Set("ssl_c3d", obj.SslC3d); err != nil {
+		return fmt.Errorf("[DEBUG] Error saving SslC3d to state for Ssl profile  (%s): %s", d.Id(), err)
+	}
+
 	if err := d.Set("ssl_forward_proxy", obj.SslForwardProxy); err != nil {
 		return fmt.Errorf("[DEBUG] Error saving SslForwardProxy to state for Ssl profile  (%s): %s", d.Id(), err)
 	}

```

**Comment by RavinderReddyF5** [comment]
@AliAllomani 
Change of Name of ssl profile is useless because it doesn't modify name of profile anyway, infact we should make it to `ForceNew` to `true` to avoid this confusion.
keeping that aside,i tried to reproduce ur issue, didnot hit any issue with change of name, i am using using latest code base.

```
bigip_ltm_profile_client_ssl.test-ClientSsl: Refreshing state... [id=/Common/test-ClientSsl]

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # bigip_ltm_profile_client_ssl.test-ClientSsl will be updated in-place
  ~ resource "bigip_ltm_profile_client_ssl" "test-ClientSsl" {
        authenticate            = "once"
        cert_extension_includes = []
        defaults_from           = "/Common/clientssl"
        id                      = "/Common/test-ClientSsl"
      ~ name                    = "/Common/test-ClientSsl" -> "/Common/test-ClientSsl-new"
        partition               = "Common"
        tm_options              = [
            "dont-insert-empty-fragments",
            "no-tlsv1.3",
        ]

        cert_key_chain {
            cert  = "/Common/default.crt"
            chain = "none"
            key   = "/Common/default.key"
            name  = "default"
        }
    }

Plan: 0 to add, 1 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

bigip_ltm_profile_client_ssl.test-ClientSsl: Modifying... [id=/Common/test-ClientSsl]
bigip_ltm_profile_client_ssl.test-ClientSsl: Modifications complete after 1s [id=/Common/test-ClientSsl]

Apply complete! Resources: 0 added, 1 changed, 0 destroyed.
```

**Comment by AliAllomani** [comment]
@RavinderReddyF5 Re-tested again, when defining bigip_ltm_profile_client_ssl only, the issue doesn't appear, however when referring it to another resource it appears.

tested with v1.1.1 and v1.9.0, same error appears

```hcl
resource "bigip_ltm_profile_client_ssl" "test-ClientSsl" {
        defaults_from           = "/Common/clientssl"
        name                    = "/Common/test-ClientSsl04"
        partition               = "Common"

}

resource "bigip_ltm_virtual_server" "vs" {
  name                       = "/Common/test-VS01"
  port                       = "443"
  destination                = "192.168.1.100"
  client_profiles            = [bigip_ltm_profile_client_ssl.test-ClientSsl.name]
}
```

Log

```hcl
bigip_ltm_profile_client_ssl.test-ClientSsl: Refreshing state... [id=/Common/test-ClientSsl03]
bigip_ltm_virtual_server.vs: Refreshing state... [id=/Common/test-VS01]

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # bigip_ltm_profile_client_ssl.test-ClientSsl will be updated in-place
  ~ resource "bigip_ltm_profile_client_ssl" "test-ClientSsl" {
        alert_timeout                       = "indefinite"
        allow_non_ssl                       = "disabled"
        authenticate                        = "once"
        authenticate_depth                  = 9
        ca_file                             = "none"
        cache_size                          = 262144
        cache_timeout                       = 3600
        cert                                = "/Common/default.crt"
        cert_extension_includes             = [
            "basic-constraints",
            "subject-alternative-name",
        ]
        cert_life_span                      = 30
        cert_lookup_by_ipaddr_port          = "disabled"
        chain                               = "none"
        ciphers                             = "DEFAULT"
        client_cert_ca                      = "none"
        crl_file                            = "none"
        defaults_from                       = "/Common/clientssl"
        forward_proxy_bypass_default_action = "intercept"
        generic_alert                       = "enabled"
        handshake_timeout                   = "10"
        id                                  = "/Common/test-ClientSsl03"
        inherit_cert_keychain               = "true"
        key                                 = "/Common/default.key"
        mod_ssl_methods                     = "disabled"
        mode                                = "enabled"
      ~ name                                = "/Common/test-ClientSsl03" -> "/Common/test-ClientSsl04"
        partition                           = "Common"
        peer_cert_mode                      = "ignore"
        proxy_ssl                           = "disabled"
        proxy_ssl_passthrough               = "disabled"
        renegotiate_period                  = "indefinite"
        renegotiate_size                    = "indefinite"
        renegotiation                       = "enabled"
        retain_certificate                  = "true"
        secure_renegotiation                = "require"
        server_name                         = "none"
        session_mirroring                   = "disabled"
        session_ticket                      = "disabled"
        sni_default                         = "false"
        sni_require                         = "false"
        ssl_forward_proxy                   = "disabled"
        ssl_forward_proxy_bypass            = "disabled"
        ssl_sign_hash                       = "any"
        strict_resume                       = "disabled"
        tm_options                          = []
        unclean_shutdown                    = "enabled"
    }

  # bigip_ltm_virtual_server.vs will be updated in-place
  ~ resource "bigip_ltm_virtual_server" "vs" {
      ~ client_profiles            = [
          - "/Common/test-ClientSsl03",
          + "/Common/test-ClientSsl04",
        ]
        destination                = "192.168.1.100"
        id                         = "/Common/test-VS01"
        ip_protocol                = "tcp"
        irules                     = []
        mask                       = "255.255.255.255"
        name                       = "/Common/test-VS01"
        persistence_profiles       = []
        policies                   = []
        port                       = 443
        profiles                   = [
            "/Common/tcp",
        ]
        server_profiles            = []
        source                     = "0.0.0.0/0"
        source_address_translation = "none"
        state                      = "enabled"
        translate_address          = "enabled"
        translate_port             = "enabled"
        vlans                      = []
        vlans_enabled              = false
    }

Plan: 0 to add, 2 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

bigip_ltm_profile_client_ssl.test-ClientSsl: Modifying... [id=/Common/test-ClientSsl03]
bigip_ltm_profile_client_ssl.test-ClientSsl: Modifications complete after 0s [id=/Common/test-ClientSsl03]

Error: Provider produced inconsistent final plan

When expanding the plan for bigip_ltm_virtual_server.vs to include new values
learned so far during apply, provider "bigip" produced an invalid new value
for .client_profiles: planned set element
cty.StringVal("/Common/test-ClientSsl04") does not correlate with any element
in actual.

This is a bug in the provider, which should be reported in the provider's own
issue tracker.
```

**Comment by RavinderReddyF5** [comment]
@AliAllomani fixed in v1.10.0, please check and open new issue if u see any problem.
